### ISSUE TITLE
ConcurrentModificationException can occur during cleaning script handling

### ISSUE THREAD

**Issue by fviale** [ow2-proactive/scheduling#2791]
The following exception can occur during cleaning script handling. A better synchronization mechanism should be implemented.
```
java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.remove(HashMap.java:1451)
	at org.ow2.proactive.scheduler.core.rmproxies.RMProxyActiveObject.cleanCallBack(RMProxyActiveObject.java:211)
	at sun.reflect.GeneratedMethodAccessor731.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.objectweb.proactive.core.mop.MethodCall.execute(MethodCall.java:353)
	at org.objectweb.proactive.core.body.request.RequestImpl.serveInternal(RequestImpl.java:214)
	at org.objectweb.proactive.core.body.request.RequestImpl.serve(RequestImpl.java:160)
	at org.objectweb.proactive.core.body.BodyImpl$ActiveLocalBodyStrategy.serveInternal(BodyImpl.java:552)
	at org.objectweb.proactive.core.body.BodyImpl$ActiveLocalBodyStrategy.serve(BodyImpl.java:485)
	at org.objectweb.proactive.core.body.AbstractBody.serve(AbstractBody.java:426)
	at org.objectweb.proactive.core.body.request.RequestReceiverImpl.receiveRequest(RequestReceiverImpl.java:167)
	at org.objectweb.proactive.core.body.BodyImpl.internalReceiveRequest(BodyImpl.java:216)
	at org.objectweb.proactive.core.body.AbstractBody.receiveRequest(AbstractBody.java:230)
	at org.objectweb.proactive.core.body.request.RequestImpl.sendRequest(RequestImpl.java:236)
	at org.objectweb.proactive.core.body.request.RequestImpl.send(RequestImpl.java:147)
	at org.objectweb.proactive.core.body.BodyImpl$ActiveLocalBodyStrategy.sendRequest(BodyImpl.java:716)
	at org.objectweb.proactive.core.body.AbstractBody.sendRequest(AbstractBody.java:446)
	at org.objectweb.proactive.core.body.future.LocalFutureUpdateCallbacks.run(LocalFutureUpdateCallbacks.java:129)
	at org.objectweb.proactive.core.body.future.FutureProxy.receiveReply(FutureProxy.java:243)
	at org.objectweb.proactive.core.body.future.FuturePool.receiveFutureValue(FuturePool.java:270)
	at org.objectweb.proactive.core.body.reply.ReplyReceiverImpl.receiveReply(ReplyReceiverImpl.java:48)
	at org.objectweb.proactive.core.body.BodyImpl.internalReceiveReply(BodyImpl.java:238)
	at org.objectweb.proactive.core.body.AbstractBody.receiveReply(AbstractBody.java:245)
	at sun.reflect.GeneratedMethodAccessor349.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.objectweb.proactive.core.mop.MethodCall.execute(MethodCall.java:353)
	at org.objectweb.proactive.core.remoteobject.RemoteObjectImpl.receiveMessage(RemoteObjectImpl.java:92)
	at org.objectweb.proactive.core.remoteobject.InternalRemoteRemoteObjectImpl.receiveMessage(InternalRemoteRemoteObjectImpl.java:134)
	at org.objectweb.proactive.extensions.pnp.PNPROMessageRequest.processMessage(PNPROMessageRequest.java:102)
	at org.objectweb.proactive.extensions.pnp.PNPServerHandler$RequestExecutor.run(PNPServerHandler.java:296)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by fviale** [comment]
The issue is critical as in that case, the nodes are not released and will remain in busy state

**Comment by fviale** [comment]
issue was fixed by commits 9054de5 and 1533b46
